### PR TITLE
Make it possible for msg.buffers in javascript to be either ArrayBuffers or views

### PR DESCRIPTION
**Pull request by jasongrout**
This makes working with message buffers in javascript a bit easier, since having views and having array buffers are both common cases.

### CODE REVIEW

**Comment by jasongrout** [comment]
@minrk or @jdfreder or @takluyver, if you have time to review this, that would be great.

File: IPython/html/static/services/kernels/serialize.js [hunk 1]
```diff
@@ -75,15 +75,17 @@ define([
         msg = _.clone(msg);
         var offsets = [];
         var buffers = [];
-        msg.buffers.map(function (buf) {
```
Comment (minrk) [inline IPython/html/static/services/kernels/serialize.js]: Why not use msg.buffers.map? The logic should be the same, yes?

File: IPython/html/static/services/kernels/serialize.js [hunk 2]
```diff
@@ -75,15 +75,17 @@ define([
...
-            buffers.push(buf);
-        });
+        var i;
+        for (i = 0; i < msg.buffers.length; i++) {
+            // msg.buffers elements could be either views or ArrayBuffers
+            // buffers elements are ArrayBuffers
+            buffers.push(msg.buffers[i].buffer || msg.buffers[i])
```
Comment (minrk) [inline IPython/html/static/services/kernels/serialize.js]: Is `.buffer` a good enough check? Or should it be an actual type check?

**Comment by minrk** [comment]
Minor questions in-line, but the principle makes sense.

File: IPython/html/static/services/kernels/serialize.js (hunk 1 above)
Comment (jasongrout) [inline IPython/html/static/services/kernels/serialize.js]: The logic is the same.  according to my trials on jsperf, `.map` is about 1/4 the speed of an explicit for loop, so I went ahead and changed it.

File: IPython/html/static/services/kernels/serialize.js (hunk 2 above)
Comment (jasongrout) [inline IPython/html/static/services/kernels/serialize.js]: All views have a .buffer attribute, and an ArrayBuffer does not.  So it handles those cases nicely.  Not all browser support all views, so I don't want to put a hardcoded list in.  But I could do something like 
```python
b = msg.buffers[i];
buffers.push(b.buffer instanceof ArrayBuffer ? b.buffer : b);
```
if you're more comfortable with that.

**Comment by jdfreder** [comment]
Looks good to me.

File: IPython/html/static/services/kernels/serialize.js (hunk 1 above)
Comment (minrk) [inline IPython/html/static/services/kernels/serialize.js]: Makes sense. Just wanted to know the reason.

**Pull request merged**